### PR TITLE
build: add ability to build vxworks in main branch

### PR DESCRIPTION
**Pull request by lt-holman**
This patch updates the source code tree to build in the main branch.
This takes some of the changes in the vxworks_layer branch, but making
the vxworks Makefile utilizing the scripts for building that are also
used by the Android build system.  This ensures that generated files are
consistent between all builds.

Signed-off-by: Keith Holman <keith.holman@windriver.com>

### CODE REVIEW

File: src/os_posix.c [hunk 1]
```diff
@@ -354,7 +328,7 @@ os_status_t os_directory_create(
 		if ( result != OS_STATUS_SUCCESS )
 		{
 			os_time_elapsed( &start_time, &time_elapsed );
-			os_time_sleep( LOOP_WAIT_TIME, OS_TRUE );
+			os_time_sleep( 100u, OS_TRUE );
```
Comment (element-82) [inline src/os_posix.c]: can you define the 100u so that it is more readable, either local scope or higher?

File: src/os_vxworks.c [hunk 2]
```diff
@@ -1,146 +1,201 @@
...
+			if ( ( *service_function )( good_argc, good_argv ) == EXIT_SUCCESS )
+				result = OS_STATUS_SUCCESS;
+
+			free( good_argv );
+		}
+	}
+	return result;
 }
 
 /** @todo fix later once it needs in vxWorks */
```
Comment (element-82) [inline src/os_vxworks.c]: might as well fix the grammar in the comment

File: src/os_vxworks.c [hunk 3]
```diff
@@ -235,76 +289,232 @@ os_status_t os_thread_rwlock_destroy(
...
+	argv[argc] = strtok (command, " ");
+	while ((argv[argc] != NULL) && (++argc < 9)) {
+		argv[argc] = strtok (NULL, " ");
+	}
+	argv[9] = NULL;
+
 	/*
 	 * Go through list of supported commands
 	 */
+#if defined(_WRS_KERNEL)
```
Comment (element-82) [inline src/os_vxworks.c]: _VXWORKS_ ?

File: src/os_vxworks.c [hunk 4]
```diff
@@ -235,76 +289,232 @@ os_status_t os_thread_rwlock_destroy(
...
+		RTP_LOADED_WAIT, VX_FP_TASK) == RTP_ID_ERROR) {
+		return OS_STATUS_FAILURE;
+	}
+	else {
+	       printf("Invalid command:%s\n", command);
+	       return OS_STATUS_FAILURE;
+	}
+#endif /* defined(_WRS_KERNEL) */
+
+#if 0
```
Comment (element-82) [inline src/os_vxworks.c]: Why is this if 0?

File: src/os_vxworks.c [hunk 5]
```diff
@@ -235,76 +289,232 @@ os_status_t os_thread_rwlock_destroy(
...
-	os_uuid_t *uuid )
+os_status_t os_system_shutdown(
+	os_bool_t reboot, unsigned int delay)
 {
-	uint32_t status;
-	os_status_t result = OS_STATUS_BAD_PARAMETER;
-
-	if ( uuid )
+	os_status_t result = OS_STATUS_FAILURE;
+#if defined(_WRS_KERNEL)
```
Comment (element-82) [inline src/os_vxworks.c]: _VXWORKS_?

**Review by element-82 — CHANGES_REQUESTED** [review]
Some comments, small things.  Otherwise looks good!

File: build-sys/vxworks-osal/Makefile [hunk 6]
```diff
@@ -0,0 +1,20 @@
...
+# distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+# WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#
+
+PRE_BUILD_DIRS = lib
+
+BUILD_DIRS = lib
+BUILD_USER_DIRS = lib
+
+include $(WIND_KRNL_MK)/rules.layers.mk
```
[the remark this answers is not in the document]
Comment (lt-holman) [reply build-sys/vxworks-osal/Makefile]: in Workbench.

File: build-sys/vxworks-osal/lib/vxworks.krnl.mak [hunk 7]
```diff
@@ -0,0 +1,59 @@
+#
```
[the remark this answers is not in the document]
Comment (lt-holman) [reply build-sys/vxworks-osal/lib/vxworks.krnl.mak]: VxWorks convention.

File: build-sys/vxworks-osal/lib/vxworks.lib.mak [hunk 8]
```diff
@@ -0,0 +1,59 @@
+#
```
[the remark this answers is not in the document]
Comment (lt-holman) [reply build-sys/vxworks-osal/lib/vxworks.lib.mak]: VxWorks convention, I think

File: build-sys/vxworks-osal/lib/vxworks.lib.mak [hunk 9]
```diff
@@ -0,0 +1,59 @@
...
+        $(DEVICE_CLOUD_OSAL_DIR)/src/uuid/unparse.c \
+        $(DEVICE_CLOUD_OSAL_DIR)/src/uuid/pack.c \
+        $(DEVICE_CLOUD_OSAL_DIR)/src/uuid/unpack.c \
+        $(DEVICE_CLOUD_OSAL_DIR)/src/uuid/randutils.c
+
+OBJS = $(addsuffix .o, $(notdir $(basename $(SRC_FILES))))
+
+EXTRA_INCLUDE += -I. -Iuuid
+
+ADDED_CFLAGS += $(C_COMPILER_FLAGS_C99)
```
[the remark this answers is not in the document]
Comment (lt-holman) [reply build-sys/vxworks-osal/lib/vxworks.lib.mak]: We compile our code to gnu99 (when using GCC).  VxWorks is replicating this setting.

File: src/os_vxworks.c (hunk 2 above)
Comment (lt-holman) [reply src/os_vxworks.c]: fixed

File: src/os_vxworks.c (hunk 3 above)
Comment (lt-holman) [reply src/os_vxworks.c]: no this is kernel specific code.

File: src/os_vxworks.c (hunk 4 above)
Comment (lt-holman) [reply src/os_vxworks.c]: mis-merged.... removed.

File: src/os_vxworks.c (hunk 5 above)
Comment (lt-holman) [reply src/os_vxworks.c]: no this is kernel specific code.

**Pull request merged**